v2/protocol/amqp/sender.go

```diff
@@ -28,7 +28,8 @@ func (s *sender) Send(ctx context.Context, in binding.Message) error {
 		return err
 	}
 
-	return s.amqp.Send(ctx, &amqpMessage)
+	err = s.amqp.Send(ctx, &amqpMessage)
```

Comment:
why this change?

Reply:
if error happened during `Send`, the `err` is used by the defer call

Reply:
oooh to access inside the Finish.

Reply:
line 22 needs the same change.